specification/dartLangSpec.tex

```diff
@@ -4427,15 +4427,18 @@ \subsection{Superinterfaces}
 When we say that a type $S$
 \IndexCustom{implements}{type!implements a type}
 another type $T$,
-this means that $T$ is a superinterface of $S$.
+this means that $T$ is a superinterface of $S$,
```

Comment:
Where do we define "is a superinterface of"?

Is `Iterable<num>` a superinterface of `List<int>`?

Reply:
> is there some existential check of "There exists a `T <: num` such that `list implements Iterable<T>`"?

I'm not sure what this is about. Considering `Iterable<num> foo<T extends List<int>>(T list) => [...list];` again, we're allowed to use `list` in the spread element because of the change that this CL makes. Then we have an expression `[...list]` of type `List<int>`. That expression is returned, the function is "normal" (non-generator and so on), and it has return type `Iterable<num>`. Assignability holds because `List<int> <: Iterable<num>`. So we only use 'implements' for the check on the spread element, and returning only involves assignability, that is: subtyping.

Reply:
The expression `[...list]` has type `List<num>` because of the context.
So, I guess I'm just checking that `<num>[...listOfInt]` is allowed. If it currently is, it will sure stay valid with this change.
And yes, reading it again, it indeed says "must implement Iterable\<*T*> for some *T*", so that's fine.

Reply:
Are you OK with this at this point?